### PR TITLE
Improves existing RP checking when creating a DB.

### PR DESCRIPTION
**Pull request by chris-ramon**
#### Details

- Currently when running an InfluxQL(`CREATE DATABASE IF NOT EXISTS xyz WITH DURATION 24h0m0s REPLICATION 1 NAME one_day_only`) to create a database which does not provide `shard duration` a default value will be set for it via:

  - [`meta.Data.CreateRetentionPolicy(...)`](https://github.com/influxdata/influxdb/blob/master/services/meta/data.go#L134) [sets](https://github.com/influxdata/influxdb/blob/master/services/meta/data.go#L145) a default value   (currently [1 hour](https://github.com/influxdata/influxdb/blob/master/services/meta/data.go#L959)).

- When re-running the same InfluxQL mentioned above, an error will occur (`retention policy conflicts with an existing policy`) when it shouldn't, based on the expected behavior of `IF NOT EXISTS` & confirmed by the [docs](https://github.com/influxdata/influxdb/blob/master/services/meta/client.go#L230):

  ```go	
// Check if the retention policy already exists. If it does and matches
// the desired retention policy, exit with no error.
```

  - So current strategy for checking retention policies does not consider the default `shard duration` value mentioned above.

- TL:DR, this PR improves the checking for existing retention policy within [`meta.Client.CreateDatabaseWithRetentionPolicy(...)`](https://github.com/influxdata/influxdb/blob/master/services/meta/client.go#L219) to consider the default `shard duration` value which is set when none is provided when creating a database.

***

- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by e-dard** [comment]
@chris-ramon thanks for this, looks good. However you'll need to move your `CHANGELOG` entry to under bugfixes for `0.13`. I suggest you rebase from master to get a more recent `CHANGELOG`.

**Comment by chris-ramon** [comment]
> @chris-ramon thanks for this, looks good. However you'll need to move your CHANGELOG entry to under bugfixes for 0.13. I suggest you rebase from master to get a more recent CHANGELOG.

Hi @fred-influx, thanks for reviewing it! - I've addressed it on 93aa287d0a9c6a4f1e4896f66f99ce8136ed8d20.

**Comment by e-dard** [comment]
LGTM :+1:
